### DISCUSSION TITLE
Doctor flagging debit entries for Expense Accounts

### DISCUSSION THREAD

**Discussion by nickdex** [ananthakumaran/paisa#92] [Q&A]
Sometimes order is cancelled or returned and it may take few days to reflect that in bank account statements which means you have two transactions on two separate days.
For refund if we don't reduce expense amount, It may inaccurately report our spending habits.
Do you have any good suggestions how to handle this case.

**Comment by ananthakumaran** [comment]
I don't have a perfect solution for this issue. I add an `Income:Refund:{name}` to `Asset:Checking` transaction to represent the refund. But, as you said, this will not reduce the expense account amount.

As for why not just move the money from expense to asset, Paisa assumes the money moves in one direction for Income and Expenses. This is true in general accounting as well. If you check out the [cashflow yearly](https://demo.paisa.fyi/cash_flow/yearly) page, the Income, and Expenses are handled differently (rolled up/down) compared to assets/liability.

I can think about adding special support for `Income:Refund:{name}`, Paisa already supports `Income:CapitalGains:{name}` convention, a similar strategy could be followed here.

**Reply by nickdex** [reply]
Hmm, I'm doing this styling of record keeping for first time so I'm not aware of conventions. Your app inspired me to finally take the plunge (have been cleaning up and merging all years for days now) Thanks!!
Coming back to topic, my main value add from this process are areas that make up my top spends and I need to plan accordingly so if any special consideration can be given for Income:Refund so that appropriate Expense account reflects reality that would be perfect.

**Reply by nickdex** [reply]
There is one more use case. When I choose  EMI plan, credit statement sometimes can take 1-2 days to convert original transaction into EMI options. Since only original item can contain itemised list you have to put expense accounts in that transaction. Credit cards handle this by reversing the original transaction just before adding first monthly EMI transaction in statement.
Let me know if this makes sense else I'll share ledger entries for more clarity

**Reply by dwaldhans** [reply]
Hello @ananthakumaran ,

I'm currently evaluating Paisa for my personal finances and such a feature would be great.

It is more or less the only thing keeping me from using Paisa as my main tool for financial analysis.

Can you give us an update if and when you are going to implement this?

Best regards,
David

**Reply by ananthakumaran** [reply]
This is something I will try to tackle for sure, but I can't say when (I don't commit to any timeline, especially for open-source projects). You can upvote the discussion, which will give some indication of how many users want this feature.

**Reply by abhishek199-dhn** [reply]
> I can think about adding special support for Income:Refund:{name}, Paisa already supports Income:CapitalGains:{name} convention, a similar strategy could be followed here.

I'm currently following this convention, but it would be great if we could add a special handling for the `Income:Refund` account.

**Reply by ayushsoni1001** [reply]
+1

**Reply by xitij2000** [reply]
I'm considering the approach mentioned in the ledger docs [here](https://ledger-cli.org/doc/ledger3.html#Tracking-reimbursable-expenses) which is to use an "Assets:Reimbursements" account for reimbursements. Likewise, I'm planning to have a Reimbursements account for reimbursements, Refunds account for Refunds and Reversal account for reversals.

**Reply by Aerics84** [reply]
Had the problem today, too.
The warning confused me. AI told me i did everything correct with my refund.

2025/05/29 * PayPal ; Gutschrift Laufschuhe
    ; PayPal Europe S.a.r.l. et Cie S.C.A
    ; Brooks
    Assets:Checking:ING                       300,00 EUR
    Expenses:Clothing

2025/05/21 * PayPal ; Laufschuhe
    ; PayPal Europe S.a.r.l. et Cie S.C.A
    ; Brooks
    Assets:Checking:ING                      -300,00 EUR
    Expenses:Clothing
